Testscripts/Windows/NET-Add-NIC-After-Provision.ps1

```diff
@@ -56,20 +56,31 @@ function Main {
 				Write-LogErr "Failed to create extra NIC #${nicNr} in $($AllVMData.ResourceGroupName)"
 				return "FAIL"
 			}
-			Start-Sleep -s 5
+			Start-Sleep -Seconds 5
 			Write-LogInfo "Successfully added extra NIC #${nicNr}!"
 		}
+		Write-LogDbg "Update VM $($AllVMData.RoleName) in RG $($AllVMData.ResourceGroupName) - start."
 		Update-AzVM -ResourceGroupName $AllVMData.ResourceGroupName -VM $vm | Out-Null
 		if (-not $?) {
 			Write-LogErr "Failed to update the VM $($AllVMData.RoleName) with new NIC(s)"
 			return "FAIL"
 		}
+		Write-LogDbg "Update VM $($AllVMData.RoleName) in RG $($AllVMData.ResourceGroupName) - end."
 		# Start VM
-		Start-AzVM -ResourceGroupName $AllVMData.ResourceGroupName -Name $AllVMData.RoleName | Out-Null
+		Write-LogDbg "Start VM $($AllVMData.RoleName) in RG $($AllVMData.ResourceGroupName)."
+		Start-AzVM -ResourceGroupName $AllVMData.ResourceGroupName -Name $AllVMData.RoleName -NoWait | Out-Null
 		if (-not $?) {
 			Write-LogErr "Failed to start $($AllVMData.RoleName)"
 			return "FAIL"
 		}
+		$vm = Get-AzVM -ResourceGroupName $AllVMData.ResourceGroupName -Name $AllVMData.RoleName -Status
+		$MaxAttempts = 20
```

Comment:
I am not sure if this works with bigger size VM provision time. Can you check 200 seconds with M416 or HB120v2 VMs?

Reply:
I am testing against bigger size, seems we need increase the max attempts.